app/src/main/java/com/example/bigowlapp/utils/AuthFailureNotificationListener.java

```diff
@@ -4,53 +4,54 @@
-                .addSnapshotListener((snapshots, e) -> {
-                    if (e != null) {
+        repositoryFacade.getCurrentUserNotificationRepository()
+                .listenToCollection((snapshots, error) -> {
+                    if (error != null) {
                         Toast.makeText(context, "BIG OWL notification listener failed: ", Toast.LENGTH_LONG).show();
                         return;
                     }
 
+                    assert snapshots != null;
                     for (DocumentChange dc : snapshots.getDocumentChanges()) {
-                        if (dc.getType().equals(ADDED)) {
-                            if (dc.getDocument().get("type").toString()
-                                    .equalsIgnoreCase("AUTH_BY_PHONE_NUMBER_FAILURE")
-                                    && dc.getDocument().getString("receiverUid")
-                                    .equalsIgnoreCase(repositoryFacade.getAuthRepository().getCurrentUser().getUid())
-                                    && !dc.getDocument().getBoolean("used")
-                                    && ((Timestamp.now().getSeconds()
-                                    - dc.getDocument().getTimestamp("creationTime").getSeconds()) < THIRTY_MINUTES)) {
-
-                                db.collection("notifications").document(dc.getDocument()
-                                        .getId()).update("used", true);
-                                String notificationSenderPhoneNum = dc.getDocument().get("senderPhoneNum").toString();
-                                String scheduleId = dc.getDocument().get("scheduleId").toString();
-                                SmsSender.sendSMS(notificationSenderPhoneNum, scheduleId);
-                            }
+                        Notification notification = NotificationRepository.getNotificationFromDocument(dc.getDocument(), Notification.class);
+                        if (!notification.isValid() || !dc.getType().equals(DocumentChange.Type.ADDED)) {
+                            continue;
+                        }
 
+                        if (notification.getType() == Notification.Type.AUTH_BY_PHONE_NUMBER_FAILURE) {
+                            handleAuthByPhoneNumberFailure((AuthByPhoneNumberFailure) notification);
                         }
                     }
                 });
+    }
+
+    private void handleAuthByPhoneNumberFailure(AuthByPhoneNumberFailure notification) {
+        if (!notification.isUsed()
+                && notification.timeSinceCreationMillis() < Schedule.MAX_TRACKING_TIME_MILLIS) {
+
+            String senderPhoneNum = notification.getSenderPhoneNum();
+            String scheduleId = notification.getScheduleId();
+            SmsSender.sendSMS(senderPhoneNum, scheduleId);
 
+            notification.setUsed(true);
+            repositoryFacade.getCurrentUserNotificationRepository()
+                    .updateDocument(notification.getUid(), notification);
+        }
     }
 }
```

Comment:
Nice way cleaning up the code and extract the logic out of all this to repository. (as mentioned above)